src/Datagrid/Pager.php

```diff
@@ -85,4 +90,35 @@ public function init()
             $this->getQuery()->setMaxResults($this->getMaxPerPage());
         }
     }
+
+    private function countCompositePrimaryKey(ProxyQueryInterface $countQuery)
```

Comment:
```suggestion
    private function countCompositePrimaryKey(ProxyQueryInterface $countQuery): void
```

Reply:
Is there any reason why we can't use `void` as return type?

Reply:
It’s strange. I added this change.